src/indexing/indexer.h

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/indexing/indexer_interface.h"
+#include "ds/logger.h"
+#include "indexing/transaction_fetcher_interface.h"
+
+#include <memory>
+#include <string>
+
+namespace ccf::indexing
+{
+  // This is responsible for managing a collection of strategies, and ensuring
+  // each has been given every transaction up to the commit point, in-order.
+  class Indexer : public IndexingStrategies
```

Comment:
I should know better than comment on naming at this point, but I do wonder if IndexBuilder rather than IndexingStrategy isn't the right descriptive and boring name here.

Reply:
The usual response: I'm sure these names are eventually wrong, but I don't know what the right names are.

I prefer `Strategy` for the current unique greppableness, both literally and mentally. The final concept likely has some intricacies/specialisations that don't quite fit initial impressions given by  `Strategy` _or_ `Builder`, but I think `Strategy` is a more sturdy term (ie - easier to remember and distinguish) on which to hang all of those extra details than `Builder` (assuming here that `IndexBuilder` is either too unwieldy to pronounce every time so becomes `Builder`, or at least that it goes into a bucket in my head keyed by `Builder`, where it is fuzzily conflated with every other `*build*` function).

Naming is hard.